crates/biome_js_analyze/src/lint/nursery/use_collapsed_if.rs

```diff
@@ -0,0 +1,214 @@
+
+        let parent_consequent = parent_if_statement.consequent().ok()?;
+        let parent_test = parent_if_statement.test().ok()?;
+        let child_consequent = child_if_statement.consequent().ok()?;
+        let child_test = child_if_statement.test().ok()?;
+
+        let parent_has_comments = match &parent_consequent {
+            AnyJsStatement::JsBlockStatement(block_stmt) => {
+                block_stmt.l_curly_token().ok()?.has_trailing_comments()
+                    || block_stmt.r_curly_token().ok()?.has_leading_comments()
+            }
+            _ => false,
+        };
+
+        let has_comments = parent_has_comments
+            || child_if_statement.syntax().has_comments_direct()
+            || child_if_statement
+                .r_paren_token()
+                .ok()?
+                .has_trailing_comments();
+        if has_comments {
+            return None;
+        }
+
+        let binary_expression = js_binary_expression(
+            parenthesized_if_needed(&parent_test),
+            token(T![&&]),
+            parenthesized_if_needed(&child_test),
+        );
+
+        // If the inner `if` statement has no block and the statement does not end with semicolon,
+        // it cannot be fixed automatically because that will break the ASI rule.
+        if !matches!(&child_consequent, AnyJsStatement::JsBlockStatement(_)) {
+            let last_token = child_consequent.syntax().last_token()?;
+            if last_token.kind() != T![;] {
+                return None;
+            }
+        }
+
+        let mut mutation = ctx.root().begin();
+        mutation.replace_node(parent_test, binary_expression.into());
+        mutation.replace_node(parent_consequent, child_consequent);
+
+        Some(JsRuleAction::new(
+            ActionCategory::QuickFix,
+            ctx.metadata().applicability(),
+            markup! { "Use collapsed "<Emphasis>"if"</Emphasis>" instead." }.to_owned(),
+            mutation,
+        ))
+    }
+}
+
+fn parenthesized_if_needed(expr: &AnyJsExpression) -> AnyJsExpression {
+    if needs_parenthesis(expr) {
+        parenthesized(expr.clone()).into()
+    } else {
+        expr.clone()
+    }
+}
+
+/// If the test expression has an operator that has lower precedence than `&&`,
+/// it needs to be wrapped with a parenthesis before concatenating expressions using `&&`.
+/// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Table
+fn needs_parenthesis(expr: &AnyJsExpression) -> bool {
+    match expr {
+        AnyJsExpression::JsLogicalExpression(expr) => matches!(
+            expr.operator().ok(),
+            Some(JsLogicalOperator::LogicalOr | JsLogicalOperator::NullishCoalescing)
+        ),
+        AnyJsExpression::JsConditionalExpression(_)
+        | AnyJsExpression::JsAssignmentExpression(_)
+        | AnyJsExpression::JsYieldExpression(_)
+        | AnyJsExpression::JsSequenceExpression(_) => true,
+        _ => false,
+    }
+}
```

Comment:
@Conaclos Thank you for your suggestion, I replaced it with `biome_js_syntax::parentheses::NeedsParentheses` but it seems not parenthesising `yield` expressions 👀 

<img width="1487" alt="image" src="https://github.com/user-attachments/assets/d8e46f01-4294-41fc-8d74-e99ae3227b41">

Reply:
Oh, I needed to check if it needs parens after building a `&&` expression. (Also I misused `JsBinaryExpression` instead of `JsLogicalExpression`. 😓)

Is this correct? 43758db

Reply:
I think so :)